pymongo/encryption.py

```diff
@@ -247,23 +247,61 @@ def close(self):
 
 
 class _Encrypter(object):
-    def __init__(self, io_callbacks, opts):
-        """Encrypts and decrypts MongoDB commands.
+    """Encrypts and decrypts MongoDB commands.
 
-        This class is used to support automatic encryption and decryption of
-        MongoDB commands.
+    This class is used to support automatic encryption and decryption of
+    MongoDB commands."""
+    def __init__(self, client, opts):
+        """Create a _Encrypter for a client.
 
         :Parameters:
-          - `io_callbacks`: A :class:`MongoCryptCallback`.
+          - `client`: The encrypted MongoClient.
           - `opts`: The encrypted client's :class:`AutoEncryptionOpts`.
         """
         if opts._schema_map is None:
             schema_map = None
         else:
             schema_map = _dict_to_bson(opts._schema_map, False, _DATA_KEY_OPTS)
+        self._bypass_auto_encryption = opts._bypass_auto_encryption
+        self._internal_client = None
+
+        def _get_internal_client(encrypter, mongo_client):
+            if encrypter._internal_client is not None:
+                return encrypter._internal_client
+            internal_client = mongo_client._duplicate(
+                minPoolSize=0, auto_encryption_opts=None)
+            encrypter._internal_client = internal_client
+            return internal_client
+
+        if opts._key_vault_client is not None:
+            key_vault_client = opts._key_vault_client
+        elif client.max_pool_size is None:
```

Comment:
I suggest refactoring to put the `client.max_pool_size is None` check inside `_get_internal_client`. This way it doesn't need to be duplicated.

Reply:
Done.